static/app/views/alerts/rules/metric/details/metricActivity.tsx

```diff
@@ -0,0 +1,239 @@
+      </div>
+      <div />
+      <div>
+        <StyledDateTime
+          date={getDynamicText({
+            value: activationActivity.dateCreated,
+            fixed: 'Mar 4, 2022 10:44:13 AM UTC',
+          })}
+          year
+          seconds
+          timeZone
+        />
+      </div>
+    </Fragment>
+  );
+}
+
+type MetricAlertActivityProps = {
+  incident: Incident;
+  organization: Organization;
+};
+
+function MetricAlertActivity({organization, incident}: MetricAlertActivityProps) {
+  // NOTE: while _possible_, we should never expect an incident to _not_ have a status_change activity
+  const activities: ActivityType[] = (incident.activities ?? []).filter(
+    activity => activity.type === IncidentActivityType.STATUS_CHANGE
+  );
+
+  const statusValues = [String(IncidentStatus.CRITICAL), String(IncidentStatus.WARNING)];
+  // TODO: kinda cheating with the forced `!`. Is there a better way to type this?
+  const latestActivity: ActivityType = activities.find(activity =>
+    statusValues.includes(String(activity.value))
+  )!;
+
+  const isCritical = Number(latestActivity.value) === IncidentStatus.CRITICAL;
+
+  // Find the _final_ most recent activity _after_ our triggered activity
+  // This exists for the `CLOSED` state (or any state NOT WARNING/CRITICAL)
+  const finalActivity = activities.find(
+    activity => activity.previousValue === latestActivity.value
+  );
+  const activityDuration = (
+    finalActivity ? moment(finalActivity.dateCreated) : moment()
+  ).diff(moment(latestActivity.dateCreated), 'milliseconds');
+
+  const triggerLabel = isCritical ? 'critical' : 'warning';
+  const curentTrigger = incident.alertRule.triggers.find(
+    trigger => trigger.label === triggerLabel
+  );
+  const timeWindow = getDuration(incident.alertRule.timeWindow * 60);
+  const alertName = capitalize(
+    AlertWizardAlertNames[getAlertTypeFromAggregateDataset(incident.alertRule)]
+  );
+
+  const project = incident.alertRule.projects[0];
+  const activation = incident.activation;
+  let activationBlock = <div />;
+  if (activation) {
+    let condition;
+    let activator;
+    switch (activation.conditionType) {
+      case String(ActivationConditionType.RELEASE_CREATION):
+        condition = 'Release';
+        activator = (
+          <GlobalSelectionLink
+            to={{
+              pathname: `/organizations/${
+                organization.slug
+              }/releases/${encodeURIComponent(activation.activator)}/`,
+              query: {project: project},
+            }}
+          >
+            {activation.activator}
+          </GlobalSelectionLink>
+        );
+        break;
+      case String(ActivationConditionType.DEPLOY_CREATION):
+        condition = 'Deploy';
+        activator = activation.activator;
+        break;
+      default:
+        condition = '--';
+    }
+    activationBlock = (
+      <div>
+        &nbsp;from {condition} {activator}
+      </div>
+    );
+  }
```

Comment:
This is such a small addition to the existing row that i actually don't think it deserves its own component 😅 
I'd need to drill the org slug and project slug too for the link... or set up contexts or something?

I dunno seems a bit much for such a small bit for now 🤷 

Here's what it looks like when there's no activations
<img width="1080" alt="Screenshot 2024-05-01 at 3 59 23 PM" src="https://github.com/getsentry/sentry/assets/6186377/3e38c47d-571d-461f-b182-954881cc6631">

Reply:
Is this the only place you need organization? if so, you could pull it out into it's own component, use the react-query hook to pull from state the org and then remove the organization passing it through a few layers of components. I think that's the kind of thing refactoring smaller components like this gets you.

I also tend to think of this refactor more as a "this will help future me", as it will give me a single place to think about this states visualization. Maybe on the list view page we could reuse the component because it's the same data model and then render it (unlikely in this case, but kinda a generalized example).

I've found that if you start to do this a bit more habitually, then everything you're building becomes a bit more extensible.

Reply:
mmm I guess I can leave a comment here to defer this.

if/when we implement more condition types, i can see a separate component being more useful to manage the `" for Release Creation/Deploy Creation/Tag Detection/Foobar"` logic
but for now, i'll keep this here to not over complicate this